src/theming.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+
+// TODO remove this after https://github.com/jxnblk/mdx-deck/pull/359
+import { useTheme } from "./use-theme";
+
+function useTokenStyles() {
+  const theme = useTheme();
+
+  const themeStylesByType = React.useMemo(() => {
+    const themeStylesByType = Object.create(null);
+    // TODO check theme.codeSurfer is defined or use default
+    theme.codeSurfer.styles.forEach(({ types, style }) => {
+      types.forEach(type => {
+        themeStylesByType[type] = Object.assign(
+          themeStylesByType[type] || {},
+          style
+        );
+      });
+    });
+    return themeStylesByType;
+  }, [theme]);
+
+  const getStyleForToken = React.useMemo(
+    () => token => {
+      return themeStylesByType[token.type] || {};
+    },
+    [themeStylesByType]
+  );
+
+  return getStyleForToken;
+}
+
+function usePreStyle() {
+  const theme = useTheme();
+  return {
+    color: theme.colors.pre || "inherit",
+    background: theme.colors.preBackround || "inherit"
```

Comment:
```diff
-preBackround
+preBackground
```

Reply:
And also, more generally, shouldn't this be `theme.pre.color` and `theme.pre.background`?

Just reading the theming docs now:

https://github.com/jxnblk/mdx-deck/blob/master/docs/theming.md#reference

I have also submitted an issue for the current pre-V2 version: https://github.com/pomber/code-surfer/issues/60

Reply:
Thanks.
I based it on the built-in themes: https://github.com/jxnblk/mdx-deck/blob/master/packages/themes/code.js#L17.  
But I'll try later what the docs say, if it works it may be better.